src/App.tsx

```diff
@@ -1,25 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedPerPageValue, setSelectedPerPageValue] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
```

Comment:
Consider using a custom hook to encapsulate the pagination logic, which includes `selectedPerPageValue`, `currentPage`, and the `handlePerPageChange` and `onPageChange` functions. This would make the `App` component cleaner and more focused on rendering.